erpnext/accounts/doctype/journal_entry/journal_entry.py

```diff
@@ -844,14 +844,14 @@ def get_outstanding(args):
 		}
 
 @frappe.whitelist()
-def get_party_account_and_balance(company, party_type, party):
+def get_party_account_and_balance(company, party_type, party, cost_center):
```

Comment:
`cost_center` should be optional

Reply:
Thank you @nabinhait for your review.
* cost_center should be optional - Sure, will make it optional.
* Cost Center should be added in Sales / Purchase Invoice for gl entry against debtors/creditors. - I thought we will be making this change as part of Branch implementation as we discussed in the meeting. I will make required change in Sales and Purchase Invoice to incorporate this change. Party GL entries will be based on item to consider multiple cost centers in one invoice.
* Cost center against party in payment entry - I have added it earlier but removed after changes suggested in #14000. I will make available again.
* Add tests - I will add required tests

Reply:
@sanjay-hercules 
>Party GL entries will be based on item to consider multiple cost centers in one invoice.

Don't split GL Entries based on multiple cost centers. For Debtors, add another field for cost center in parent (sales invoice).

> Cost center against party in payment entry - I have added it earlier but removed after changes suggested in #14000. I will make available again.

In your last implementation, I think main issue was introducing 2 fields `cost_center_from` and `cost_center_to` in Payment Entry. I think we can add one field in PE. And for the scenario, where we need 2 cost centers, we can manage those entries via Journal Entry.

Reply:
>Don't split GL Entries based on multiple cost centers. For Debtors, add another field for cost center in parent (sales invoice).

So you are now okay to add cost center field in parent document which will get enabled only if **maintain cost center in entry of balance sheet account** option is checked in Account Settings. We need to do same in Purchase Invoice. **It would be better to consider Delivery Note and Purchase Receipt documents as well for cost center.**

>I think we can add one field in PE. And for the scenario, where we need 2 cost centers, we can manage those entries via Journal Entry.

Sure, I will